src/lib/server/hyrise_communicator.cpp

```diff
@@ -0,0 +1,111 @@
+                                                                   const std::string& query) {
+  // Named prepared statements must be explicitly closed before they can be redefined by another Parse message
+  // https://www.postgresql.org/docs/10/static/protocol-flow.html
+  if (Hyrise::get().storage_manager.has_prepared_plan(statement_name)) {
+    AssertInput(statement_name.empty(),
+                "Named prepared statements must be explicitly closed before they can be redefined.");
+    Hyrise::get().storage_manager.drop_prepared_plan(statement_name);
+  }
+
+  auto pipeline_statement = SQLPipelineBuilder{query}.create_pipeline_statement();
+  auto sql_translator = SQLTranslator{UseMvcc::Yes};
+  auto prepared_plans = std::vector<std::shared_ptr<AbstractLQPNode>>();
+  try {
+    prepared_plans = sql_translator.translate_parser_result(*pipeline_statement.get_parsed_sql_statement());
+  } catch (const InvalidInputException& exception) {
+    return std::string(exception.what());
+  }
+  Assert(prepared_plans.size() == 1u, "Only a single statement allowed in prepared statement");
+
+  const auto prepared_plan =
+      std::make_shared<PreparedPlan>(prepared_plans[0], sql_translator.parameter_ids_of_value_placeholders());
+
+  Hyrise::get().storage_manager.add_prepared_plan(statement_name, std::move(prepared_plan));
+  return {};
+}
+
+std::pair<std::string, std::shared_ptr<AbstractOperator>> HyriseCommunicator::bind_prepared_plan(
+    const PreparedStatementDetails& statement_details) {
+  Assert(Hyrise::get().storage_manager.has_prepared_plan(statement_details.statement_name),
+         "The specified statement does not exist.");
+
+  const auto prepared_plan = Hyrise::get().storage_manager.get_prepared_plan(statement_details.statement_name);
+
+  if (statement_details.statement_name.empty()) {
+    Hyrise::get().storage_manager.drop_prepared_plan(statement_details.statement_name);
+  }
+
+  auto parameter_expressions = std::vector<std::shared_ptr<AbstractExpression>>{statement_details.parameters.size()};
+  for (auto parameter_idx = size_t{0}; parameter_idx < statement_details.parameters.size(); ++parameter_idx) {
+    parameter_expressions[parameter_idx] =
+        std::make_shared<ValueExpression>(statement_details.parameters[parameter_idx]);
+  }
+
+  std::shared_ptr<AbstractOperator> pqp;
+  std::string error = "";
+  try {
+    const auto lqp = prepared_plan->instantiate(parameter_expressions);
+    pqp = LQPTranslator{}.translate_node(lqp);
+  } catch (const std::exception& exception) {
+    error = exception.what();
+  }
+
+  return std::make_pair(error, pqp);
+}
+
+std::shared_ptr<TransactionContext> HyriseCommunicator::get_new_transaction_context() {
+  return Hyrise::get().transaction_manager.new_transaction_context();
+}
+
+std::shared_ptr<const Table> HyriseCommunicator::execute_prepared_statement(
+    const std::shared_ptr<AbstractOperator>& physical_plan) {
+  std::shared_ptr<const Table> result_table;
```

Comment:
104 & 105: shouldn’t the compiler complain about unused variables?

Reply:
I have no clue why ...

1. this is still there
2. the compiler isn't complaining